### ISSUE TITLE
Use Zinc for dependency checking

### ISSUE THREAD

**Issue by jadenPete** [bazelbuild/rules_scala#1668]
**This issue is part of a broader proposal to merge some of the features in [lucidsoftware/rules_scala](https://github.com/lucidsoftware/rules_scala) into this ruleset.**

[The tracking issue for this effort](https://github.com/bazelbuild/rules_scala/issues/1658)

[The relevant proposal section](https://docs.google.com/document/d/12BmUPpJpworA9ep2P_J00TxXwFRBZDQttUCsoYo0Yao/edit?tab=t.0#heading=h.ry919yzcfl1l).

If we [compiled with Zinc](https://github.com/bazelbuild/rules_scala/issues/1665), we could use Zinc's analysis file to check for unused dependencies and used transitive dependencies not declared as direct dependencies. You can read the details of the proposed architecture, its pros and cons, and a comparison between it and the existing architecture [here](https://docs.google.com/document/d/12BmUPpJpworA9ep2P_J00TxXwFRBZDQttUCsoYo0Yao/edit?tab=t.0#heading=h.1vcoe25mzvu5).

TL;DR is that by using Zinc for dependency checking, we can:
- Greatly simplify this repository's dependency checking code
- No longer have to [patch the compiler](https://github.com/bazelbuild/rules_scala/tree/master/dt_patches) to report which depedendencies were used. This should enable any compiler to be used with the ruleset, regardless of this repository's support for it
- No longer have to rely on a [compiler plugin](https://github.com/bazelbuild/rules_scala/pull/243) for dependency checking
- Move dependency checking to a worker and make it a [validation action](https://bazel.build/extending/rules#validation_actions), speeding up compilation
